### PR TITLE
feat(Select): add disabled option

### PR DESCRIPTION
**Pull request by teimurjan**
[FX-2160](https://toptal-core.atlassian.net/browse/FX-2160)

[Abstract](https://share.goabstract.com/b2f57088-6713-47f1-a4fd-2f3f60a9d648?mode=design&sha=3674422f1a9915b54bb950319af52986416c311c)

### Description

- Added a disabled option.
- Fixed Sidebar disabled items' colors
  - [Abstract (light)](https://share.goabstract.com/3d7f8be3-40f1-48a0-933e-b80929348cfc?mode=design&sha=77ad4a4f04f395e22be2a1309c60385c2be6784b)
  - [Abstract (dark)](https://share.goabstract.com/dda60d94-5f1a-4ed8-895d-44d6ec5bd116?mode=design&sha=77ad4a4f04f395e22be2a1309c60385c2be6784b)

### How to test

1. Go to the Select storybook page.
2. Open the disabled options example.
3. Click on the select.
4. Hover over a disabled option.
5. Be sure that it can't be highlighted.
6. Click on a disabled option.
7. Be sure that it's not selected.
8. Highlight a non-disabled option.
9. Press arrow down and arrow up multiple times.
10. Be sure that the disabled items are skipped.

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| Insert screenshots or screen recordings | Insert screenshots or screen recordings |

### Review

- [ ] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/api-principles.md)
- [ ] Annotate all `props` in component with documentation
- [ ] Create `examples` for component
- [ ] Ensure that deployed demo has expected results and good examples
- [ ] Ensure that tests pass by running `yarn test`
- [ ] Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- [ ] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

### CODE REVIEW

**Review by vshyrokov — CHANGES_REQUESTED** [review]
Please fix bugs. Videos attahced.

**Comment by teimurjan** [comment]
@vshyrokov Great catch! Fixed ✅

**Comment by vshyrokov** [comment]
The color of the disabled item is still incorrect because we have an opacity of 0.5 there.

File: packages/picasso/src/Select/types.ts [hunk 1]
```diff
@@ -129,7 +130,6 @@ export interface UseSelectStateProps {
 }
 
 export type UseSelectStateOutput = {
-  selectedIndexes: number[]
```
[the remark this answers is not in the document]
Comment (teimurjan) [reply packages/picasso/src/Select/types.ts:132]: It's not used anywhere.

File: packages/picasso/src/Select/hooks/use-highlighted-index/test.ts [hunk 2]
```diff
@@ -1,36 +1,145 @@
...
+      setHighlightedIndex(result, 4)
     })
-    expect(result.current[0]).toBe(2)
+    expect(getHighlightIndex(result)).toBe(4)
 
     rerender(false)
-    expect(result.current[0]).toBe(0)
+    expect(getHighlightIndex(result)).toBe(1)
   })
 })
```
Comment (vshyrokov) [inline packages/picasso/src/Select/hooks/use-highlighted-index/test.ts:145]: it seems we are missing tests when we have a Select with group options, WDYT?

**Pull request merged**